lib/screens_web/controllers/v2/screen_controller.ex

```diff
@@ -83,41 +83,81 @@ defmodule ScreensWeb.V2.ScreenController do
 
     config = Cache.screen(screen_id)
 
-    case config do
-      %Screen{app_id: app_id} ->
-        refresh_rate = Parameters.get_refresh_rate(app_id)
+    if is_struct(config, Screen) do
+      assigns = get_assigns(params, screen_id, config)
 
-        conn
-        |> assign(:app_id, app_id)
-        |> assign(:refresh_rate, refresh_rate)
-        |> assign(:audio_readout_interval, Parameters.get_audio_readout_interval(app_id))
-        |> assign(
-          :audio_interval_offset_seconds,
-          Parameters.get_audio_interval_offset_seconds(config)
-        )
-        |> assign(:sentry_frontend_dsn, Application.get_env(:screens, :sentry_frontend_dsn))
-        |> assign(
-          :refresh_rate_offset,
-          calculate_refresh_rate_offset(screen_id, refresh_rate)
-        )
-        |> assign(:is_real_screen, match?(%{"is_real_screen" => "true"}, params))
-        |> assign(:screen_side, screen_side(params))
-        |> assign(:requestor, params["requestor"])
-        |> assign(:disable_sentry, params["disable_sentry"])
-        |> assign(:rotation_index, rotation_index(params))
-        |> assign(:triptych_pane, triptych_pane(params))
-        |> put_view(ScreensWeb.V2.ScreenView)
-        |> render("index.html")
-
-      nil ->
-        render_not_found(conn)
+      conn
+      |> merge_assigns(assigns)
+      |> put_view(ScreensWeb.V2.ScreenView)
+      |> render("index.html")
+    else
+      render_not_found(conn)
     end
   end
 
   def index(conn, _params) do
     render_not_found(conn)
   end
 
+  def index_pending(conn, %{"id" => screen_id} = params) do
+    # Should we log info for these requests?
+    # Current precedent is to not, for simulation requests.
+    # But it could be helpful, and we don't have to come up with a way to
+    # distinguish between requests from "real screens" vs not.
+    # (Because real screens should never hit this endpoint.)
+    # _ = Screens.LogScreenData.log_pending_screen_page_load(screen_id)
```

Comment:
Oop I left this code comment that's more like a PR comment/question, want me to remove it?

Reply:
Ohhhhhh I forgot to comment on this, also in the PR description. I think you're wise to point out the logging potential, and might be worth adding here and now

Reply:
Actually, I've let the Screenplay PRs back up a ton because I was so slow to review. Let's remember this as a #todo and count it as a follow-on item?